DHT sensor configured via port expander (sx1509) errors out during build

### The problem

I'm having a SX1509 GPIO expander configured via which I'm trying to configure a DHT22 temp/humid sensor.
The config file check passes, though during building the firmware the process errors out with:

```src/main.cpp: In function 'void setup()':
src/main.cpp:256:40: error: no matching function for call to 'esphome::dht::DHT::set_pin(esphome::sx1509::SX1509GPIOPin*&)'
   dht_dht->set_pin(sx1509_sx1509gpiopin);
                                        ^
src/main.cpp:256:40: note: candidate is:
In file included from src/esphome.h:2:0,
                 from src/main.cpp:3:
src/esphome/components/dht/dht.h:39:8: note: void esphome::dht::DHT::set_pin(esphome::InternalGPIOPin*)
   void set_pin(InternalGPIOPin *pin) { pin_ = pin; }
        ^
src/esphome/components/dht/dht.h:39:8: note:   no known conversion for argument 1 from 'esphome::sx1509::SX1509GPIOPin*' to 'esphome::InternalGPIOPin*'
```

### Which version of ESPHome has the issue?

2021.10.2

### What type of installation are you using?

pip

### Which version of Home Assistant has the issue?

_No response_

### What platform are you using?

ESP8266

### Board

_No response_

### Component causing the issue

sensor:dht - sx1509

### Example YAML snippet

```yaml
sx1509:
  - id: sx1509_hub
    address: 0x3E

sensor:
  - platform: dht
    #model: AM2302
    pin:
      sx1509: sx1509_hub
      number: 8
    temperature:
      name: "temp"
    humidity:
      name: "humid"
    update_interval: 10s
```


### Anything in the logs that might be useful for us?

```txt
src/main.cpp: In function 'void setup()':
src/main.cpp:256:40: error: no matching function for call to 'esphome::dht::DHT::set_pin(esphome::sx1509::SX1509GPIOPin*&)'
   dht_dht->set_pin(sx1509_sx1509gpiopin);
                                        ^
src/main.cpp:256:40: note: candidate is:
In file included from src/esphome.h:2:0,
                 from src/main.cpp:3:
src/esphome/components/dht/dht.h:39:8: note: void esphome::dht::DHT::set_pin(esphome::InternalGPIOPin*)
   void set_pin(InternalGPIOPin *pin) { pin_ = pin; }
        ^
src/esphome/components/dht/dht.h:39:8: note:   no known conversion for argument 1 from 'esphome::sx1509::SX1509GPIOPin*' to 'esphome::InternalGPIOPin*'
```


### Additional information

_No response_

--- a/esphome/components/dht/sensor.py
+++ b/esphome/components/dht/sensor.py
@@ -33,7 +33,7 @@ DHT = dht_ns.class_("DHT", cg.PollingComponent)
 CONFIG_SCHEMA = cv.Schema(
     {
         cv.GenerateID(): cv.declare_id(DHT),
-        cv.Required(CONF_PIN): pins.gpio_input_pin_schema,
+        cv.Required(CONF_PIN): pins.internal_gpio_input_pin_schema,
         cv.Optional(CONF_TEMPERATURE): sensor.sensor_schema(
             unit_of_measurement=UNIT_CELSIUS,
             accuracy_decimals=1,

